src/main/java/gov/nasa/pds/api/engineering/controllers/MyCollectionsApiController.java

```diff
@@ -125,68 +114,64 @@ public MyCollectionsApiController(ObjectMapper objectMapper, HttpServletRequest
     		
 	
     
-    @SuppressWarnings("unchecked")
 	private Products getProductChildren(String lidvid, int start, int limit, List<String> fields, List<String> sort, boolean onlySummary) throws IOException, LidVidNotFoundException {
-  	
     	if (!lidvid.contains("::")) lidvid = this.productBO.getLatestLidVidFromLid(lidvid);
     	MyCollectionsApiController.log.info("request collection lidvid, collections children: " + lidvid);
-         
-    	try {
-	    	Products products = new Products();
-	    	
-	    	HashSet<String> uniqueProperties = new HashSet<String>();
-	    	
-	      	Summary summary = new Summary();
-	    	
-	    	summary.setStart(start);
-	    	summary.setLimit(limit);
-	    	
-	    	if (sort == null) {
-	    		sort = Arrays.asList();
-	    	}	
-	    	summary.setSort(sort);
-	    	
-	    	products.setSummary(summary);
-	    	
-	    	CollectionProductRefBusinessObject  collectionProductRefBO = new CollectionProductRefBusinessObject(this.esRegistryConnection);
-	    	CollectionProductRelationships collectionProductRelationships = collectionProductRefBO.getCollectionProductsIterable(lidvid, start, limit);
-	    	
-	    	for (EntityProduct eProd : collectionProductRelationships) {
-				  if (eProd != null) {
-		        	MyCollectionsApiController.log.info("request lidvdid: " + eProd.getLidVid() );
-		        	
-		    		Product product = ElasticSearchUtil.ESentityProductToAPIProduct(eProd, this.getBaseURL());
-				    		
-		    		Map<String, XMLMashallableProperyValue> filteredMapJsonProperties = ProductBusinessObject.getFilteredProperties(
-		    				eProd.getProperties(), 
-		    				fields,
-		    				null);
-		
-		    		uniqueProperties.addAll(filteredMapJsonProperties.keySet());
-		
-		    		if (!onlySummary) {
-		        		product.setProperties((Map<String, PropertyArrayValues>)(Map<String, ?>)filteredMapJsonProperties);
-		        		
-		        		products.addDataItem(product);
-		    		}
+
+    	int iteration=0,wsize=0;
+    	HashSet<String> uniqueProperties = new HashSet<String>();
+    	List<String> plidvids = new ArrayList<String>();
+    	List<String> wlidvids = new ArrayList<String>();
+    	Products products = new Products();
+      	Summary summary = new Summary();
+
+    	if (sort == null) { sort = Arrays.asList(); }	
+
+    	summary.setStart(start);
+    	summary.setLimit(limit);
+      	summary.setSort(sort);	
+    	products.setSummary(summary);
+
+    	for (final Map<String,Object> kvp : new ElasticSearchHitIterator(10, this.esRegistryConnection.getRestHighLevelClient(),
```

Comment:
Why size is 10 here ? Does it matter ? For performance I guess...

Reply:
Oh yes, this really caused me some problems. I had it using the limit given by the user as with the other calls and it had strange performance implications. While running the notebook, noticed that the limit of 500 caused faster data transfers but long duration between blocks downloaded. Tracked it down elasticsearch took much longer to get 100 lidvids than 10 than linear transfer would suggest. 10 was the fastest (also elasticsearch default) that I found in the 3 attempts at debugging.

The difference with the bundle is that the ref_lid_collection item is usually short (not thousand) so that limit use alright with using the user limit.